### DISCUSSION TITLE
Windows PowerShell is in NonInteractive mode. Read and Prompt functionality is not available.

### DISCUSSION THREAD

**Discussion by dima-b** [microsoft/PSRule#1439] [Q&A]
HI, could you help me to figure out the reason of the error below:

**##[error]Exception calling "ShouldContinue" with "2" argument(s): "Windows PowerShell is in NonInteractive mode. Read and Prompt functionality is not available."**

Starting: Analyze Azure template files
==============================================================================
Task         : PSRule analysis
Description  : Run analysis with PSRule.
Version      : 2.6.2301031
Author       : Bernie White
Help         : https://github.com/Microsoft/PSRule-pipelines/blob/main/docs/tasks.md#ps-rule-assert
==============================================================================
[info] Source 'D:\azure-agent\_work\7\s\.ps-rule\' does not exist.

Checking PSRule
[info] Using repository: PSGallery
[info] Installing PSRule: 2.7.0
##[error]Exception calling "ShouldContinue" with "2" argument(s): "Windows PowerShell is in NonInteractive mode. Read and Prompt functionality is not available."
Finishing: Analyze Azure template files

**Comment by BernieWhite** [comment]
@dima-b Thanks for the question. The error is cause by the pipeline attempting to switch into a mode where it prompts to continue, this is not a feature in a pipeline as there is no way to provide input while the pipeline is running.

This is not a normal behaviour in my experience, so it could indicate a problem with your configuration or a bug.

Let me ask a few additional questions to see if we can work out the issue:

Are you able to provide a snippet of your pipeline YAML code?
Are you running on a self-hosted agent?

**Comment by dima-b** [comment]
Hi @BernieWhite thank you for the answer. Yes, you are right I'm running on a self-hosted agent.
Below is a snippet of my pipeline, but I guess I was figured out how to solve the problem. In my case I manually installed PSRule module on the agents and it started to work without errors. I believe it might be related to a missing NuGet plugin on the newly installed agent. If it is a true it worth mentioning in the install guide for psrule. I will try to reproduce the issue on a new agent and let you know.
     - task: ps-rule-assert@2
          displayName: Analyze Azure template files
          continueOnError: true
          inputs:
            modules: 'PSRule.Rules.Azure,PSRule.Monitor'
            baseline: 'Azure.GA_2022_12'
            inputPath: '' #$(pathToBicepFile)
            outputFormat: NUnit3
            outputPath: $(psRuleResultsPath)

**Reply by BernieWhite** [reply]
@dima-b Good point, could be as these are not installed within Windows PowerShell by default and it will prompt to install the NuGet provider if it is not installed or installed with `-Force`.

**Reply by BernieWhite** [reply]
Reference docs now over here: https://microsoft.github.io/PSRule/v2/troubleshooting/#windows-powershell-is-in-noninteractive-mode

@dima-b If your suggestion answered your own question please mark as answer so that it might help someone in the community. Thanks.

Otherwise let us know.
